### PR TITLE
Cycle Sort In Ruby

### LINKED ISSUE #2146: Cycle Sort
- [ ] C - @saloniankita
- [x] C++
- [x] C#
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [x] Go - @m-code12
- [x] Java
- [x] JS - @JanaSabuj
- [ ] Kotlin
- [ ] Php
- [x] Python
- [x] Readme
- [x] Ruby - @HardevKhandhar
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by HardevKhandhar**
Fixes #2146 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Cycle Sort In Ruby
- 

#### Languages Used:
- Ruby
- 

#### Files Added:
- Cycle_Sort.rb
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Cycle_Sort/Cycle_Sort.rb [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+puts 
+
+puts "Input Array: #{input_array}"
+puts "Cycle Sort: "
+
+cycleSort!(input_array)
+puts "#{input_array}"
+
+= begin
```
Comment (atarax665) [inline Cycle_Sort/Cycle_Sort.rb:66]: Remove the space after =

File: Cycle_Sort/Cycle_Sort.rb [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
```
Comment (atarax665) [inline Cycle_Sort/Cycle_Sort.rb:67]: Mention Sample Input heading here

File: Cycle_Sort/Cycle_Sort.rb [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+Enter the value at index 6: 1
+Enter the value at index 7: 9
+Enter the value at index 8: 3.5
+Enter the value at index 9: 5
+Enter the value at index 10: 8
+Enter the value at index 11: 4
+Enter the value at index 12: 7
+Enter the value at index 13: 0
+Enter the value at index 14: 6
+
```
Comment (atarax665) [inline Cycle_Sort/Cycle_Sort.rb:85]: Mention Sample Output heading

File: Cycle_Sort/Cycle_Sort.rb [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+Input Array: [0, 1, 2, 2, 2, 2, 1, 9, 3, 5, 8, 4, 7, 0, 6]
+Cycle Sort: [0, 0, 1, 1, 2, 2, 2, 2, 3, 4, 5, 6, 7, 8, 9]
+
+= end
```
Comment (atarax665) [inline Cycle_Sort/Cycle_Sort.rb:89]: Remove space after =

**Comment by atarax665** [comment]
Looks Good now. Please squash your commits.

**Comment by HardevKhandhar** [comment]
Dear Sir @atarax665 
The suggested changes have been done.
The commits are squashed.
Thank you.

**Pull request merged**